src/tests/ascent/t_ascent_commands.cpp

```diff
@@ -0,0 +1,473 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) Lawrence Livermore National Security, LLC and other Ascent
+// Project developers. See top-level LICENSE AND COPYRIGHT files for dates and
+// other details. No copyright assignment is required to contribute to Ascent.
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+//-----------------------------------------------------------------------------
+///
+/// file: t_ascent_cinema_a.cpp
+///
+//-----------------------------------------------------------------------------
+
+#include "gtest/gtest.h"
+
+#include <ascent.hpp>
+
+#include <iostream>
+#include <math.h>
+
+#include <conduit_blueprint.hpp>
+
+#include "t_config.hpp"
+#include "t_utils.hpp"
+
+
+
+using namespace std;
+using namespace conduit;
+using namespace ascent;
+
+index_t EXAMPLE_MESH_SIDE_DIM = 32;
+bool example_bool = false;
+
+//-----------------------------------------------------------------------------
+void void_callback_1(conduit::Node &params, conduit::Node &output)
+{
+    output["param_was_passed"] = false;
+    if (params.has_path("example_param"))
+    {
+        output["param_was_passed"] = true;
+    }
+}
+
+//-----------------------------------------------------------------------------
+void void_callback_2(conduit::Node &params, conduit::Node &output)
+{
+    if (example_bool)
+    {
+        example_bool = false;
+    }
+    else
+    {
+        example_bool = true;
+    }
+}
+
+//-----------------------------------------------------------------------------
+bool bool_callback_1()
+{
+    return example_bool;
+}
+
+//-----------------------------------------------------------------------------
+bool bool_callback_2()
+{
+    return !example_bool;
+}
+
+//-----------------------------------------------------------------------------
+TEST(ascent_commands, register_no_name_void)
+{
+    //
+    // Run Ascent
+    //
+    Ascent ascent;
+    Node ascent_opts;
+    // default is now ascent
+    ascent_opts["runtime/type"] = "ascent";
+    ascent.open(ascent_opts);
+
+    // An error should be thrown due to not including a name
+    ascent::register_callback("", void_callback_1);
```

Comment:
we can wrap this with a google test, `EXPECT_THROW` to make sure an error is thrown

```suggestion
    EXPECT_THROW(ascent::register_callback("", void_callback_1),conduit::Error);
```